### PR TITLE
Add diagnostics for single file legacy mocking

### PR DESCRIPTION
**Pull request by gayaldassanayake**
## Purpose
$ subject

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/39643

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by gayaldassanayake** [comment]
Other than the fix for the issue, this PR contains additional improvements for `InvalidFunctionMockingTestCase`. 

However, need to change the structure of the tests to get the expected output from a file instead of a string once https://github.com/ballerina-platform/ballerina-lang/pull/39562 is merged.

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java [hunk 1]
```diff
@@ -179,6 +179,13 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
             String functionName = functionNode.getName().getValue();
 
             if (MOCK_ANNOTATION_NAME.equals(annotationName)) {
+                // Check if the package belongs to a single file project
+                if (packageName.equals(DOT)) {
```
[the remark this answers is not in the document]
Comment (Dilhasha) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java:183]: Shall we also loom at renaming the `DOT` constant with some other name such as `PACKAGE_NAME_SEPARATOR`?

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java [hunk 2]
```diff
@@ -179,6 +179,13 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
             String functionName = functionNode.getName().getValue();
 
             if (MOCK_ANNOTATION_NAME.equals(annotationName)) {
+                // Check if the package belongs to a single file project
+                if (SINGLE_FILE_PACKAGE_NAME.equals(packageName)) {
+                    diagnosticLog.logDiagnostic(
+                            DiagnosticSeverity.ERROR, (ModuleDescriptor) null, attachmentNode.getPosition(),
+                            "function mocking is not supported with standalone Ballerina files");
```
Comment (azinneera) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java:186]:
```suggestion
                            "compile-time function mocking is not supported with standalone Ballerina files");
```

Comment (gayaldassanayake) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/MockAnnotationProcessor.java:186]: Should the error message for both legacy and new function mocking be changed to match this?

**Review by azinneera — APPROVED** [review]

**Pull request merged**